### DISCUSSION TITLE
Java 11 or 17?

### DISCUSSION THREAD

**Discussion by hvbtup** [eclipse-birt/birt#1077] [Q&A] [answered]
In some of the build-related discussions, I saw that Java 11 was replaced with Java 17.
Does this mean that Java 17 is needed for *building* BIRT 4.10 now?
Is Java 11 still sufficient for the runtime?

**Comment by claesrosell** [comment, marked as answer]
JavaSE-11 is still used build-time and runtime, If someone is using Java17-SE during development, that is ok as long as only Java11-SE features is used. I think that the ExecutionEnvironment is set to JavaSE-11 on all our bundles.

**Reply by wimjongman** [reply]
I agree. I have included Java 17 as the runtime in our products. However, we still build against java 11.

**Comment by hvbtup** [comment]
Ok then, this comment on https://github.com/eclipse/birt/commit/6dd4b76d1134ca9dfd67eaeda66a076a4b75b9eb confused me:
Created an alternative configuration for the RCP designer
Move embedded jre to 17

**Reply by wimjongman** [reply]
We ship the BIRT products with a Java runtime so that user don't have to install Java. This runtime was updated from V11 to V17
